src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -47,21 +53,18 @@ public ModelStateDictionary(int maxAllowedErrors)
         /// </summary>
         /// <param name="dictionary">The <see cref="ModelStateDictionary"/> to copy values from.</param>
         public ModelStateDictionary(ModelStateDictionary dictionary)
+            : this(dictionary?.MaxAllowedErrors ?? DefaultMaxAllowedErrors)
```

Comment:
Yes, which is why I asked the follow-up below. NRE here doesn't sound like a big deal and it would have been fine if you set `MaxAllowedErrors` as before anyhow. Easiest approach would seem to be a `private` constructor, inline initializer or `Init()` method to handle `_root`.